File: 프로그래머스/3/12979. 기지국 설치/기지국 설치.java
class Solution {
    public int solution(int n, int[] stations, int w) {
        int i;
        int len;
        int cnt;
        int range;

        range = w << 1 | 1;
        len = stations.length;
        cnt = 0;
        if (stations[0] - w > 1) {
            cnt += (stations[0] - w - 2) / range + 1;
        }
        for (i = 1; i < len; i++) {
            if (stations[i - 1] + (w << 1) + 1 < stations[i]) {
                cnt += (stations[i] - stations[i - 1] - range - 1) / range + 1;
            }
        }
        if (stations[len - 1] + w < n) {
            cnt += (n - stations[len - 1] - w - 1) / range + 1;
        }
        return cnt;
    }
}

File: 프로그래머스/3/12979. 기지국 설치/기지국 설치.cpp
#include <iostream>
#include <vector>

using namespace std;

int solution(int n, vector<int> stations, int w) {
    int i;
    int cnt;
    int gap;
    int size;
    int range;

    range = w << 1 | 1;
    size = stations.size();
    cnt = 0;
    if ((gap = stations[0] - w - 1) > 0) {
        cnt += (gap - 1) / range + 1;
    }
    for (i = 1; i < size; i++) {
        if ((gap = stations[i] - stations[i - 1] - range) > 0) {
            cnt += (gap - 1) / range + 1;
        }
    }
    if ((gap = n - stations[size - 1] - w) > 0) {
        cnt += (gap - 1) / range + 1;
    }
    return cnt;
}